streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -237,13 +237,15 @@ private static boolean isRecoverable(final KafkaException uncaughtException) {
      * @throws TaskMigratedException
      */
     protected void commitTransaction(final Map<TopicPartition, OffsetAndMetadata> offsets,
-                           final ConsumerGroupMetadata consumerGroupMetadata) {
+                                     final ConsumerGroupMetadata consumerGroupMetadata) {
         if (!eosEnabled()) {
             throw new IllegalStateException(formatException("Exactly-once is not enabled"));
         }
         maybeBeginTransaction();
         try {
-            producer.sendOffsetsToTransaction(offsets, consumerGroupMetadata);
+            // Older brokers don't understand any group metadata beyond the group id, thus we must downgrade the request for eos-v1
```

Comment:
nit: maybe clarify a bit better here, e.g. "EXACTLY_ONCE_V2 on the Streams config assumes that broker is at least 2.5 and hence understand ... ; if it is smaller than V2, then the brokers may be on older versions and hence ..."

Reply:
Ack